### PR TITLE
Stats: fetching clicks / referrer

### PR DESCRIPTION
**Pull request by jklausa**
### Description

Adds support for fetching clicks and referrer data.

They're so similar in shape I took the liberty of grouping them both in one PR

### Testing Details

* Verify the code builds
* Verify that the tests pass

- [x] Please check here if your pull request includes additional test coverage.

### CODE REVIEW

File: WordPressKit/Time-based data/ClicksStatsType.swift [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        } else {
+            children = []
+        }
+
+        let icon = jsonDictionary["icon"] as? String
+        let urlString = jsonDictionary["url"] as? String
+
+        self.title = title
+        self.clicksCount = clicksCount
+        self.clickedURL = urlString.flatMap { URL(string: $0) }
```
Comment (stevebaranski) [inline WordPressKit/Time-based data/ClicksStatsType.swift]: I recognize why we'd use flatMap on a collection. It appears that the benefit in this case is to handle the optionality of `icon` & `urlString`, correct?

File: WordPressKit/Time-based data/ReferrerStatsType.swift [hunk 2]
```diff
@@ -0,0 +1,84 @@
+public struct ReferrerStatsType {
+    public let period: StatsPeriodUnit
+    public let periodEndDate: Date
+
+    public let totalReferrerViews: Int
```
Comment (stevebaranski) [inline WordPressKit/Time-based data/ReferrerStatsType.swift]: In `ClicksStatsType`, there appears to be an analogous property called `totalClicksCount`. Is this worth renaming to `totalViewsCount` or `totalRefererredViewsCount` for consistency with that and `StatsReferrer` (`viewsCount`) below?

File: WordPressKit/Time-based data/ReferrerStatsType.swift [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    public let otherReferrerViews: Int
```
Comment (stevebaranski) [inline WordPressKit/Time-based data/ReferrerStatsType.swift]: If we embrace the suggestion above, it is also applicable here.

**Comment by stevebaranski** [comment]
@jklausa I just noted that the hash in `Podfile.lock` was updated when I ran `pod install` to review this branch. If you find that's the case for you, would you please include a commit for that too?

File: WordPressKit/Time-based data/ClicksStatsType.swift (hunk 1 above)
Comment (jklausa) [reply WordPressKit/Time-based data/ClicksStatsType.swift]: Exactly! It saves us doing something like: 
```swift
let clickedURL: URL?
if let urlString = urlString {
    clickedURL = URL(string: urlString)
} else {
    clickedURL = nil
}
```

Just a bit of syntactic sugar.

File: WordPressKit/Time-based data/ReferrerStatsType.swift (hunk 2 above)
Comment (jklausa) [reply WordPressKit/Time-based data/ReferrerStatsType.swift]: Yup, absolutely! Thanks for catching that!

**Pull request merged**